tests/phpunit/tests/blocks/wpBlock.php

```diff
@@ -352,6 +362,292 @@ public function test_render_applies_dynamic_render_block_filter() {
+			),
+		);
+	}
+
+	/**
+	 * @ticket 63676
+	 * @covers WP_Block::render()
+	 *
+	 * @dataProvider data_provider_test_render_enqueues_scripts_and_styles
+	 *
+	 * @param Closure|null $set_up
+	 * @param string       $block_markup
+	 * @param string[]     $expected_styles
+	 * @param string[]     $expected_scripts
+	 * @param string[]     $expected_script_modules
+	 */
+	public function test_render_enqueues_scripts_and_styles( ?Closure $set_up, string $block_markup, string $expected_rendered_block, array $expected_styles, array $expected_scripts, array $expected_script_modules ) {
+		if ( $set_up instanceof Closure ) {
+			$set_up();
+		}
+		wp_register_style( 'static-view-style', home_url( '/static-view-style.css' ) );
+		wp_register_script( 'static-view-script', home_url( '/static-view-script.js' ) );
+		wp_register_script_module( 'static-view-script-module', home_url( '/static-view-script-module.js' ) );
+		$this->registry->register(
+			'core/static',
+			array(
+				'view_style_handles'     => array( 'static-view-style' ),
+				'view_script_handles'    => array( 'static-view-script' ),
+				'view_script_module_ids' => array( 'static-view-script-module' ),
+			)
+		);
+
+		wp_register_style( 'static-child-view-style', home_url( '/static-child-view-style.css' ) );
+		wp_register_script( 'static-child-view-script', home_url( '/static-child-view-script.js' ) );
+		wp_register_script_module( 'static-child-view-script-module', home_url( '/static-child-view-script-module.js' ) );
+		$this->registry->register(
+			'core/static-child',
+			array(
+				'view_style_handles'     => array( 'static-child-view-style' ),
+				'view_script_handles'    => array( 'static-child-view-script' ),
+				'view_script_module_ids' => array( 'static-child-view-script-module' ),
+			)
+		);
+
+		wp_register_style( 'dynamic-view-style', home_url( '/dynamic-view-style.css' ) );
+		wp_register_script( 'dynamic-view-script', home_url( '/dynamic-view-script.js' ) );
+		wp_register_script_module( 'dynamic-view-script-module', home_url( '/dynamic-view-script-module.js' ) );
+		$this->registry->register(
+			'core/dynamic',
+			array(
+				'render_callback'        => static function () {
+					return '<p class="dynamic">Hello World!</p>';
+				},
+				'view_style_handles'     => array( 'dynamic-view-style' ),
+				'view_script_handles'    => array( 'dynamic-view-script' ),
+				'view_script_module_ids' => array( 'dynamic-view-script-module' ),
+			)
+		);
+
+		// TODO: Why not use do_blocks() instead?
+		$parsed_blocks  = parse_blocks( trim( $block_markup ) );
+		$parsed_block   = $parsed_blocks[0];
+		$context        = array();
+		$block          = new WP_Block( $parsed_block, $context, $this->registry );
+		$rendered_block = $block->render();
+
+		$this->assertEqualHTML(
+			$expected_rendered_block,
+			$rendered_block,
+			'<body>',
+			"Snapshot:\n$rendered_block"
```

Comment:
This isn't a clear error message. 

```suggestion
			'Rendered block does not contain expected HTML'
```

Reply:
The snapshot is included so you can easily update the test case. Let's keep that:

```suggestion
			"Rendered block does not contain expected HTML:\n$rendered_block"
```